server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailQuery.scala

```diff
@@ -37,16 +38,50 @@ object Limit {
 }
 case class QueryState(value: String) extends AnyVal
 
+sealed trait SortProperty {
+  def toSortClause: SortClause
+}
+case object ReceivedAtSortProperty extends SortProperty {
+  override def toSortClause: SortClause = SortClause.Arrival
+}
+
+object IsAscending {
+  val DESCENDING: IsAscending = IsAscending(false)
+}
+case class IsAscending(sortByASC: Boolean) extends AnyVal {
+  def toSortOrder: SearchQuery.Sort.Order = if (sortByASC) SearchQuery.Sort.Order.NATURAL else SearchQuery.Sort.Order.REVERSE
+
+}
+
+object Comparator {
+  val default: Comparator = Comparator(ReceivedAtSortProperty, Some(IsAscending.DESCENDING), None)
+}
+
+case class Collation(value: String) extends AnyVal
+
+case class Comparator(property: SortProperty,
+                      isAscending: Option[IsAscending],
+                      collation: Option[Collation]) {
+  def toSort: SearchQuery.Sort = new SearchQuery.Sort(property.toSortClause, isAscending.getOrElse(IsAscending.DESCENDING).toSortOrder)
```

Comment:
Same here, default should be true for `isAcending`

Reply:
Here I agree